src/classes/adapter.js

```diff
@@ -0,0 +1,144 @@
+const Base = require('./base')
+const Knex = require('knex')
+
+module.exports = class Adapter extends Base {
+  /**
+   * Sets up the knex query builder instance
+   * @constructor
+   */
+  constructor (config) {
+    super()
+    this.config = config
+    this.knex = Knex({
+      client: 'pg',
+      connection: config,
+      searchPath: 'knex,public'
+    })
+  }
+
+  /**
+   * Converts a models attributes hash into an array of column names for use in
+   * an sql query
+   *
+   * @param {Object} attributes - attributes hash of the form
+   * @example {title: 'string', createdAt: 'date'}
+   * @return {Array} - an array of column names
+   */
+  fieldsFromAttributes (attributes) {
+    return Object.keys(attributes)
+  }
+
+  parseIncludes () {
+
+  }
+
+  include () {
+
+  }
+
+  /**
+   * Finds all records for a given model. @see `query` when you need more control over
+   * what records are returned.
+   *
+   * @param {Object} Model - object with properties `tableName` and an `attributes` hash
+   * @return {Array} - array of database records
+   */
+  findAll (Model) {
+    const attributes = this.fieldsFromAttributes(Model.definition.attributes)
+    const tableName = Model.tableName
+    return this.knex.column(attributes).select().from(tableName)
+  }
+
+  /*
+  options should be implemented here to jsonapi spec
+  {
+    include: 'comments, posts, comments.author',
+    fields: {
+      articles: 'title, body',
+      people: 'name'
+    },
+    sort: 'age, -name',
+    page: {number: 1, size: 20},
+    filter: {}
+  }
+  */
+  query (Model, options) {
+    const attributes = this.fieldsFromAttributes(Model.definition.attributes)
+    const tableName = Model.tableName
+
+    if (typeof options.fields === 'object') {
+      const types = Object.keys(options.fields)
+      if (types.indexOf(Model.type) !== -1) {
+        const fields = options.fields[Model.type].split(',').map(field => field.trim())
+        for (const attr of attributes) {
+          if (fields.indexOf(attr) === -1) {
+            attributes.splice(attributes.indexOf(attr))
+          }
+        }
+        attributes.push('id')
+      }
+    }
+
+    let query = this.knex.column(attributes).select().from(tableName)
+
+    if (typeof options.page === 'object' && options.page.number && options.page.size) {
```

Comment:
would prefer to see defaults here instead of forcing the user to define both (instead of letting them define just one). If you define neither, you get defaults anyways, right?

```js
if (typeof options.page === 'object' && (options.page.number || options.page.size)) {
  const pageNumber = options.page.number || 0
  const page = options.page.size || 20 // or pull that default from some config or something
  // ...
} 
```